client/consensus/manual-seal/src/seal_block.rs

```diff
@@ -153,9 +153,14 @@ pub async fn seal_block<B, BI, SC, C, E, TP, CIDP>(
 			digest_provider.append_block_import(&parent, &mut params, &inherent_data)?;
 		}
 
+		// Make sure we return the same post-hash that will be calculated when importing the block
+		// This is important in case the digest_provider added any signature, seal, ect.
+		let mut post_header = header.clone();
+		post_header.digest_mut().logs.extend(params.post_digests.iter().cloned());
```

Comment:
wondering if there's any security risk to calculating the hash up front at this point rather than cloning the header and doing the hash later on?

Reply:
Why should that be a security risk?